### PR TITLE
Fix broken canary container Dockerfile

### PR DESCRIPTION
**Pull request by RedbackThomson**
The canary Dockerfile was failing due to the necessary `software-properties-common` package not being installed before calling `add-apt-repository`. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/images/Dockerfile.canary [hunk 1]
```diff
@@ -7,11 +7,11 @@ RUN apt-get update && apt-get install -y curl \
     python-pip \
     vim \
     sudo
-
+
```
Comment (cadedaniel) [inline tests/images/Dockerfile.canary]: nit: remove unnecessary whitespace

File: tests/images/Dockerfile.canary [hunk 2]
```diff
@@ -7,11 +7,11 @@ RUN apt-get update && apt-get install -y curl \
     python-pip \
     vim \
     sudo
-
+    
 RUN pip install awscli
 
 # Install yq
-RUN sudo add-apt-repository ppa:rmescandon/yq && apt update && apt install -y yq
+RUN apt-get update && apt install -y apt-transport-https ca-certificates curl gnupg-agent software-properties-common && sudo add-apt-repository ppa:rmescandon/yq && apt update && apt install -y yq
```
Comment (cadedaniel) [inline tests/images/Dockerfile.canary]: Why was this necessary? Can you update comment?

Comment (RedbackThomson) [reply tests/images/Dockerfile.canary]: In body of pull request - you need `software-properties-common` to use the `add-apt-repository` command.

Comment (cadedaniel) [reply tests/images/Dockerfile.canary]: Ah sorry I missed. Weird that it is failing now.

Can you still update yq comment?

Comment (cadedaniel) [reply tests/images/Dockerfile.canary]: ignore I saw outdated change
